app/src/main/java/com/babylon/wallet/android/domain/common/Result.kt

```diff
@@ -1,64 +1,11 @@
 package com.babylon.wallet.android.domain.common
 
-// Temporary solution until we replace our own Result to Kotlin's Result
-typealias KotlinResult<T> = kotlin.Result<T>
-
-sealed interface Result<out T> {
-
-    data class Success<T>(val data: T) : Result<T>
-
-    data class Error(
-        val exception: Throwable? = null
-    ) : Result<Nothing>
-}
-
-suspend fun <T> Result<T>.onValue(action: suspend (T) -> Unit) = apply {
-    if (this is Result.Success) {
-        action(this.data)
-    }
-}
-
-suspend fun <T, R> Result<T>.map(action: suspend (T) -> R): Result<R> {
-    return when (this) {
-        is Result.Error -> Result.Error(this.exception)
-        is Result.Success -> Result.Success(action(data))
-    }
-}
-
-suspend fun <T, R> Result<Collection<T>>.mapIfNotEmpty(action: suspend (Collection<T>) -> R): Result<R> {
-    return when (this) {
-        is Result.Error -> Result.Error(this.exception)
-        is Result.Success -> {
-            if (data.isEmpty()) {
-                Result.Error(Exception("Empty collection"))
-            } else {
-                Result.Success(action(data))
-            }
-        }
-    }
-}
-
-inline fun <T, R> Result<T>.switchMap(action: (T) -> Result<R>): Result<R> {
-    return when (this) {
-        is Result.Error -> Result.Error(this.exception)
-        is Result.Success -> action(this.data)
-    }
-}
-
-fun <T> Result<T>.value(): T? {
-    if (this is Result.Success) {
-        return data
-    }
-    return null
-}
-
-suspend fun <T> Result<T>.onError(action: suspend (Throwable?) -> Unit) = apply {
-    if (this is Result.Error) {
-        action(this.exception)
-    }
-}
-
-fun <T> Result<T>.asKotlinResult(): KotlinResult<T> = when (this) {
-    is Result.Success -> KotlinResult.success(value = data)
-    is Result.Error -> KotlinResult.failure(exception = exception ?: error(""))
-}
+suspend fun <T, R> Result<T>.map(action: suspend (T) -> R): Result<R> = fold(
```

Comment:
All we do in both cases if using fold so let me know if we want to keep it or remove those extensions and use fold wherever needed instead.

Reply:
what do you mean here? 👀 
can you give an example?

Reply:
So all this **map** and **switchMap** methods do is wrap **fold** .
I thought if it does not make sense we could ditch both map and switchMap and use fold respectively but it might introduce more code being copied over etc. So up to you.

Reply:
There is a map method with contracts in place https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/map.html. I don't think we have to use our own implementation.

I believe since map exists, and also since the cases were switch map is used are less and not so common is to use the already defined map function by kotlin and discard switch map.